### PR TITLE
[rllib] Port QMIX, MADDPG to new execution API

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This ports the last remaining algorithms before we can deprecate policy optimizers.

### CODE REVIEW

File: rllib/execution/replay_ops.py [hunk 1]
```diff
@@ -91,3 +91,74 @@ def gen_replay(_):
                 yield item
 
     return LocalIterator(gen_replay, SharedMetrics())
+
+
+class MixInReplay:
```
Comment (ericl) [inline rllib/execution/replay_ops.py:96]: Note that this operator is copied from IMPALA PR not yet merged.

File: rllib/contrib/maddpg/maddpg.py [hunk 2]
```diff
@@ -171,10 +171,28 @@ def collect_metrics(trainer):
     return result
 
 
+def add_maddpg_postprocessing(config):
+    """Add the before learn on batch hook.
+
+    This hook is called explicitly prior to TrainOneStep() in the execution
+    setups for DQN and APEX.
```
Comment (ericl) [inline rllib/contrib/maddpg/maddpg.py:178]: It's a bit unfortunate we have to add this hook, but the alternative to define a new execution plan just for MADDPG is probably too messy.

Comment (sven1977) [reply rllib/contrib/maddpg/maddpg.py:178]: Yeah, probably not necessary right now.

**Review by sven1977 — APPROVED** [review]
Btw, could we add some example scripts at some point to show, how each of these single(!) ops and classes can work in dummy agents? That would be really helpful.

**Comment by ericl** [comment]
You mean exercising the ops in standalone examples / unit-test style to show what they do?

**Pull request merged**